protocol/x/clob/keeper/msg_server_cancel_orders_test.go

```diff
@@ -55,6 +57,46 @@ func TestCancelOrder_PanicIfValidationSucceedsButOrderNotFound(t *testing.T) {
 	)
 }
 
+func TestCancelOrder_InfoLogIfOrderNotFound(t *testing.T) {
+	memClob := &mocks.MemClob{}
+	memClob.On("SetClobKeeper", mock.Anything).Return()
+	ks := keepertest.NewClobKeepersTestContext(t, memClob, &mocks.BankKeeper{}, &mocks.IndexerEventManager{})
+	msgServer := keeper.NewMsgServerImpl(ks.ClobKeeper)
+
+	orderToCancel := constants.CancelLongTermOrder_Alice_Num0_Id0_Clob0_GTBT15
+
+	ctx := ks.Ctx.WithBlockHeight(2)
+	mockLogger := &mocks.Logger{}
+	mockLogger.On("With", mock.Anything, mock.Anything).Return(mockLogger)
+	mockLogger.On("Info",
+		errorsmod.Wrap(
+			errorsmod.Wrapf(
+				types.ErrStatefulOrderDoesNotExist,
+				"Order Id to cancel does not exist. OrderId : %+v",
+				orderToCancel.OrderId,
+			),
+			"MsgCancelOrder failed DeliverTx because the order was already removed from state",
+		).Error(),
+	).Return()
+	ctx = ctx.WithLogger(mockLogger)
+	ctx = ctx.WithBlockTime(time.Unix(int64(2), 0))
+	ks.BlockTimeKeeper.SetPreviousBlockInfo(ctx, &blocktimetypes.BlockInfo{
+		Height:    2,
+		Timestamp: time.Unix(int64(2), 0),
+	})
+
+	// MsgCancelOrder will fail because the order is not in state, but it should not log an error because
+	// because the order is present in ProcessProposerMatchesEvents.RemovedStatefulOrderIds
+	ks.ClobKeeper.MustSetProcessProposerMatchesEvents(
+		ctx,
+		types.ProcessProposerMatchesEvents{BlockHeight: 2, RemovedStatefulOrderIds: []types.OrderId{orderToCancel.OrderId}},
+	)
+
+	_, err := msgServer.CancelOrder(ctx, &orderToCancel)
+	require.ErrorIs(t, err, types.ErrStatefulOrderDoesNotExist)
+	mockLogger.AssertExpectations(t)
+}
+
 func TestCancelOrder_Error(t *testing.T) {
```

Comment:
Suggestion: add a test case where `ExpectedError` is something different than `types.ErrStatefulOrderDoesNotExist`.